src/compiler/factory/nodeFactory.ts

```diff
@@ -2801,6 +2801,9 @@ namespace ts {
             else if (isLogicalOrCoalescingAssignmentOperator(operatorKind)) {
                 node.transformFlags |= TransformFlags.ContainsES2021;
             }
+            else if (operatorKind === SyntaxKind.InKeyword && isPrivateIdentifier(left)) {
+                node.transformFlags |= TransformFlags.ContainsESNext;

```

Comment:
Why does this need `ContainsESNext`? A `PrivateIdentifier` is already marked with `ContainsClassFields` and the transformation happens in the `classFields` transformer.

To be honest, `ContainsESNext` isn't actually a useful flag (and one we should remove) since it doesn't actually do anything in the `esnext` transformer.

Reply:
Good point, I removed it and all seems fine. Thanks for spotting that.